src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/DntrTemplateReplacementService.kt

```diff
@@ -0,0 +1,32 @@
+package uk.gov.justice.digital.hmpps.makerecalldecisionapi.service
+
+import com.deepoove.poi.XWPFTemplate
+import org.springframework.core.io.ClassPathResource
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.makerecalldecisionapi.domain.makerecalldecisions.recommendation.DntrData
+import uk.gov.justice.digital.hmpps.makerecalldecisionapi.jpa.entity.RecommendationEntity
+import uk.gov.justice.digital.hmpps.makerecalldecisionapi.util.MrdTextConstants.Constants.EMPTY_STRING
+import uk.gov.justice.digital.hmpps.makerecalldecisionapi.util.RecommendationToDocumentMapper
+
+@Service
+internal class DntrTemplateReplacementService {
+  val resource = ClassPathResource("DNTR Template.docx")
+
+  fun generateDocFromTemplate(recommendation: RecommendationEntity): String {
```

Comment:
This looks near identical to the one found in `PartATemplateReplacementService` - can we make a generic class/interface that sits over these ones?

Reply:
I had a go, but found myself forking with if statements on the output filenames and struggled generifying the mappings. I could use the strategy pattern encapsulate them both into a single dependency for the RecommendationService to use. But, that doesn't really reduce the code footprint, and I feel it's overkill as we only have 2 strategies (document types).

I am totally open to suggestion on this point :)

Reply:
I had a go, but found myself needing to pass in the filename/data which blurred the single responsibility of the RecommendationService. I also struggled with generifying the mappings. 

I could use the strategy pattern encapsulate them both into a single dependency for the RecommendationService to use. But, that doesn't really reduce the code footprint, and I feel it's overkill as we only have 2 strategies (document types).

I am totally open to suggestion on this point :)

Reply:
Ok maybe let's get it merged and we can refactor it at some point